pandas/_libs/tslibs/period.pyx

```diff
@@ -1453,8 +1453,8 @@ def extract_ordinals(object[:] values, freq):
             try:
                 ordinals[i] = p.ordinal
 
-                if p.freqstr != freqstr:
-                    msg = DIFFERENT_FREQ_INDEX.format(freqstr, p.freqstr)
+                if p.freqstr != freq.freqstr:
+                    msg = DIFFERENT_FREQ_INDEX.format(freq.freqstr, p.freqstr)
```

Comment:
everything here is orthogonal to the rest of the PR, right?

Reply:
Yeah, this is insignificant from a performance perspective - just related cleanup.